node/network/gossip-support/src/lib.rs

```diff
@@ -313,10 +330,26 @@ impl State {
 
 		// issue another request for the same session
 		// if at least a third of the authorities were not resolved
-		self.last_failure = if failures >= num / 3 {
-			Some(Instant::now())
+		if failures >= num / 3 {
+			let timestamp = Instant::now();
+			match self.failure_start {
+				None => self.failure_start = Some(timestamp),
+				Some(first) if first.elapsed() >= LOW_CONNECTIVITY_WARN_DELAY => {
+					tracing::warn!(
+						target: LOG_TARGET,
+						connected = ?(num - failures),
+						target = ?num,
+						"Low connectivity"
+					);
+
+				}
+				Some(_) => {}
+			}
```

Comment:
IMHO this is still weird.

When you create a new request and the first fails, you store `failure_start`. Then it will take 10 minutes until you report something. But if the first request already failed 10 minutes after the node started, why not print the message directly? What should change in these extra 10 minutes?

Reply:
Why would the first failure occur 10 minutes after startup? We immediately start trying to connect to all validators and will retry approximately on every slot. So for this to happen we would need to have been able to connect to all validators immediately on startup and then later (after 10 minutes) suddenly loose connectivity.  We could complicate matters a bit and have the 10 minute delay on startup and some smaller one during normal operation, but I don't think it's worth it.

And low connectivity in the first 10 minutes after the node has started is expected, so I don't want any warnings there.

Reply:
What I mean is the following:

1. Startup. We start to connect to all validators.
2. This works after some minutes. So, nothing else happens.
3. New session, we start to connect to the new validators.
4. This time it fails on the first request. With your implementation nothing would happen, but I'm asking why not print here directly because the node is already running for quite some time and authority discovery should be up and running.

Reply:
Got it, so you are worried that the 10 minute timeout is too long for issues during operation after startup. 

Which makes me realize that this 2/3 threshold is not correctly calculated anyways. We are connecting to all validators of the current, the previous and the next session (or even more past sessions). So in theory the lookup could fail for all validators of the current session and we would still not print any warning. :scream: 

But ignoring this problem for now. If we get a new validator set of lets say 800 validators, would the lookup then not indeed take again 8+ minutes? I mean we would likely not get a warning, because of the total number of validators we already did lookups for (current and past sessions), but this is kind of broken behavior.

It is starting to feel like this is just an additional safe guard at best and a proper solution should be implemented in terms of [actual connection tracking](https://github.com/paritytech/polkadot/issues/3410) where we can also map connections to actual sessions and issue connection warnings per individual sessions.

For gossip support we should re-evaluate whether stopping of retries after the 2/3rd threshold is a good idea though, as this is not per session.

Reply:
> But ignoring this problem for now. If we get a new validator set of lets say 800 validators, would the lookup then not indeed take again 8+ minutes? I mean we would likely not get a warning, because of the total number of validators we already did lookups for (current and past sessions), but this is kind of broken behavior.

I believe we start querying the cache at the same time it starts getting populated, because we try to connect to the same set of authorities `AuthorityDiscoveryApi::authorities`.

Reply:
Yeah could be, so yeah, we should ignore what I said ;P